core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -615,6 +618,10 @@ private DefaultFunctionResolver date_format() {
     );
   }
 
+  private ExprValue dayOfMonthToday(Clock clock) {
+    return new ExprIntegerValue(formatNow(clock).getDayOfMonth());
```

Comment:
```suggestion
    return new ExprIntegerValue(LocalDateTime.now(clock).getDayOfMonth());
```

Reply:
Addressed in 469fbb23ed6003676fa19f98c4c80e838ba3d949

Reply:
Are you missing a commit or it may be reverted?